pythonx/jedi_vim.py

```diff
@@ -157,49 +157,84 @@ def wrapper(*args, **kwargs):
     return func_receiver
 
 
+# XXX: might use lru cache, given that you can configure it per buffer?!
 current_environment = (None, None)
 
 
+def set_environment(executable=None, bufnr=None):
+    global current_environment
+
+    # print(repr(executable), repr(bufnr))
+
+    if not executable:  # "unset"
+        executable = 'auto'
+        environment = None
+    else:
+        try:
+            # XXX: create_environment is not deterministic
+            #      (https://github.com/davidhalter/jedi/issues/1155)
+            # jedi.create_environment(executable, safe=False)
+            environment = jedi.api.environment.Environment(executable)
```

Comment:
Just for information, I had to comment this line and use `environment = jedi.create_environment(executable, safe=False)` instead to get it working.

I'm using python3.7 with pyenv.

Reply:
@Fenkiou 
Thanks for trying this.
Did you get an error before? ("'executable=%s is not supported: %s.'")
If so, what was the message?

Reply:
Nope the message was this one:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/alex/.vim/repos/github.com/davidhalter/jedi-vim/pythonx/jedi_vim.py", line 177, in set_environment
    environment = jedi.api.environment.Environment(executable=executable)
TypeError: __init__() missing 1 required positional argument: 'path'
```

But in fact, I didn't run `git submodule update` when checking out your branch, everything is working correctly without any change! Sorry for the noise.